compiler/src/dotty/tools/dotc/config/CompilerCommand.scala

```diff
@@ -69,14 +67,22 @@ object CompilerCommand {
   def checkUsage(summary: ArgsSummary, sourcesRequired: Boolean)(using Context): List[String] = {
     val settings = ctx.settings
 
-    /** Creates a help message for a subset of options based on cond;
-     *  terminalWidth = 0 means no terminal is available.
-     */
-    def availableOptionsMsg(cond: Setting[?] => Boolean, terminalWidth: Int = 0): String = {
+    /** Creates a help message for a subset of options based on cond */
+    def availableOptionsMsg(cond: Setting[?] => Boolean): String = {
       val ss = (settings.allSettings filter cond).toList sortBy (_.name)
       val maxNameWidth = 30
       val nameWidths = ss.map(_.name.length).partition(_ < maxNameWidth)._1
-      val width = if nameWidths.nonEmpty then nameWidths.max else maxNameWidth 
+      val width = if nameWidths.nonEmpty then nameWidths.max else maxNameWidth
+      val terminalWidth =
+        val pageWidth = settings.pageWidth.value
+        val columnsVar = System.getenv("COLUMNS")
+        if columnsVar != null then columnsVar.toInt
+        else if Properties.isWin then
+          val ansiconVar = System.getenv("ANSICON") // eg. "142x32766 (142x26)"
```

Comment:
Checking these environment variables seems like a good idea, but instead of doing it here, we could use them to set a default value for the `pagewidth` setting instead of `80`, that way we'll use them in other situations where we're printing things: https://github.com/lampepfl/dotty/blob/6b9796c38c37957c6a6dba387c4a820b06976356/compiler/src/dotty/tools/dotc/config/ScalaSettings.scala#L44